deno_dist/utils/types.ts

```diff
@@ -10,3 +10,21 @@ export type UnionToIntersection<U> = (U extends any ? (k: U) => void : never) ex
 ) => void
   ? I
   : never
+
+export type JSONPrimitive = string | boolean | number | null | undefined
+export type JSONArray = (JSONPrimitive | JSONObject | JSONArray)[]
+export type JSONObject = { [key: string]: JSONPrimitive | JSONArray | JSONObject }
+export type JSONValue = JSONObject | JSONArray | JSONPrimitive
+
+// `boolean` will be `true` or `false` because it's an alias of them.
+// See: https://github.com/microsoft/TypeScript/issues/22596
+// This type converts `true | false` to `boolean` that we expect.
+type TrueAndFalseToBoolean<T> = T extends true ? boolean : T extends false ? boolean : T
```

Comment:
@yudai-nkt 

I added the comment and made `TrueAndFalseToBoolean` that is more descriptive. As noted here, `boolean` is distributed `true | false`, so I'm fixing it. I'm sure there are other ways to do it, but this is how I did it.